api/v1alpha1/apidefinition_types.go

```diff
@@ -418,26 +418,74 @@ type APIDefinitionSpec struct {
 	//UpstreamCertificates       map[string]string     `json:"upstream_certificates"`
 	//PinnedPublicKeys           map[string]string     `json:"pinned_public_keys"`
 
-	// EnableJWT enables JSON web token authentication
+	// EnableJWT set JWT as the access method for this API.
 	EnableJWT bool `json:"enable_jwt,omitempty"`
 	//UseGoPluginAuth            bool                  `json:"use_go_plugin_auth"`
 
 	EnableCoProcessAuth bool `json:"enable_coprocess_auth,omitempty"`
 
 	// JWTSigningMethod algorithm used to sign jwt token
 	// +kubebuilder:validation:Enum=rsa;hmac;ecdsa
-	JWTSigningMethod           string            `json:"jwt_signing_method,omitempty"`
-	JWTSource                  string            `json:"jwt_source,omitempty"`
-	JWTIdentityBaseField       string            `json:"jwt_identity_base_field,omitempty"`
-	JWTClientIDBaseField       string            `json:"jwt_client_base_field,omitempty"`
-	JWTPolicyFieldName         string            `json:"jwt_policy_field_name,omitempty"`
-	JWTDefaultPolicies         []string          `json:"jwt_default_policies,omitempty"`
-	JWTIssuedAtValidationSkew  uint64            `json:"jwt_issued_at_validation_skew,omitempty"`
-	JWTExpiresAtValidationSkew uint64            `json:"jwt_expires_at_validation_skew,omitempty"`
-	JWTNotBeforeValidationSkew uint64            `json:"jwt_not_before_validation_skew,omitempty"`
-	JWTSkipKid                 bool              `json:"jwt_skip_kid,omitempty"`
-	JWTScopeToPolicyMapping    map[string]string `json:"jwt_scope_to_policy_mapping,omitempty"`
-	JWTScopeClaimName          string            `json:"jwt_scope_claim_name,omitempty"`
+	JWTSigningMethod string `json:"jwt_signing_method,omitempty"`
+
+	// JWTSource Must either be a base64 encoded valid RSA/HMAC key or a url to a
+	// resource serving JWK, this key will then be used to validate inbound JWT and
+	// throttle them according to the centralised JWT options and fields set in the
+	// configuration.
+	JWTSource string `json:"jwt_source,omitempty"`
```

Comment:
The JWT source does not need to be b64 encoded I don't think. Needs checking.

Reply:
This must be `base64` encoded, only when it is a `jwk` url, then it can either be `base64` or plain url string.